presto-hive/src/main/java/com/facebook/presto/hive/HiveMetastoreClientFactory.java

```diff
@@ -66,28 +67,45 @@ public HiveMetastoreClient create(String host, int port)
     protected TTransport createTransport(String host, int port)
             throws TTransportException
     {
-        TTransport transport;
         if (socksProxy == null) {
-            transport = new TTransportWrapper(new TSocket(host, port, timeoutMillis), host);
-            transport.open();
-        }
-        else {
-            Socket socks = createSocksSocket(socksProxy);
+            TTransport transport = new TTransportWrapper(new TSocket(host, port, timeoutMillis), host);
             try {
-                socks.connect(InetSocketAddress.createUnresolved(host, port), timeoutMillis);
-                socks.setSoTimeout(timeoutMillis);
-            }
-            catch (IOException e) {
-                throw rewriteException(new TTransportException(e), host);
-            }
-            try {
-                transport = new TTransportWrapper(new TSocket(socks), host);
+                transport.open();
+                return transport;
             }
             catch (TTransportException e) {
```

Comment:
This only closes when a `TTransportException` is throw.  Instead, catch `Throwable` and re-throw.